backend/sites/src/app/services/site/site.service.ts

```diff
@@ -238,4 +271,120 @@ export class SiteService {
+      async (transactionalEntityManager: EntityManager) => {
+        try {
+          if (sitesSummary) {            
+            sitesSummary.srStatus = SRApprovalStatusEnum.Pending;
+            await transactionalEntityManager.save(Sites, sitesSummary);
+          } else {
+            console.log('No changes To Site Summary');
+          }
+
+          if (events) {
+
+            events.map(event=> event.srAction = SRApprovalStatusEnum.Pending);
+
+            transactionalEntityManager.save(Events, events);
+            
+          } else {
+            console.log('No changes To Site Events');
+          }
+
+          if (eventsParticipants) {
+
+            eventsParticipants.map(participant => participant.srAction = SRApprovalStatusEnum.Pending);
+
+            await transactionalEntityManager.save(
+              EventPartics,
+              eventsParticipants,
+            );
+          } else {
+            console.log('No changes To Site Event Participants');
+          }
+
+          if (siteParticipants) {
+            siteParticipants.map(participant => participant.srAction = SRApprovalStatusEnum.Pending);
+            await transactionalEntityManager.save(
+              SitePartics,
+              siteParticipants,
+            );
+          } else {
+            console.log('No changes To Site Participants');
+          }
+
+          if (siteAssociations) {
+            siteAssociations.map(association => association.srAction = SRApprovalStatusEnum.Pending);
+            await transactionalEntityManager.save(SiteAssocs, siteAssociations);
+          } else {
+            console.log('No changes To Site Associations');
+          }
+
+          if (subDivisions) {
+            subDivisions.map(subDivison => subDivison.srAction = SRApprovalStatusEnum.Pending);
+            await transactionalEntityManager.save(Subdivisions, subDivisions);
+          } else {
+            console.log('No changes To Site subDivisions');
+          }
+
+          if (landHistories) {
+            landHistories.map(history => history.srAction = SRApprovalStatusEnum.Pending);
+            await transactionalEntityManager.save(LandHistories, landHistories);
+          } else {
+            console.log('No changes To Site LandHistories');
+          }
+
+          if (profiles) {
+            profiles.map(profile => profile.srAction = SRApprovalStatusEnum.Pending);
+            await transactionalEntityManager.save(SiteProfiles, profiles);
+          } else {
+            console.log('No changes To Site profiles');
+          }
+
+
+          const historyLog:HistoryLog = {
+            userId : userInfo??userInfo.sub,
+            content : inputDTO,
+            id: null,
+            whoCreated: userInfo??userInfo.givenName,
+            whenCreated: new Date(),
+            whenUpdated : new Date(),
+            whoUpdated: userInfo??userInfo.givenName,
+          }
+
+          this.historyLogRepository.save(historyLog);
```

Comment:
should we have an await? and assign result of await to a variable and handle it

Reply:
done added await